lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
+          unless what.is_a? String
+            if klass.to_s.classify.constantize.find_by_id(what.id)
```

Comment:
Not sure I understand the point of this line - isn't `what` an instance of a model already?